addons/cds-connector-0.0.1/chart/cds-connector/templates/gw.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: kubeless.io/v1beta1
+kind: Function
+metadata:
+  annotations:
+    function-size: S
+  name: cds-events
+spec:
+  checksum: ""
+  deployment:
+    metadata:
+      creationTimestamp: null
+    spec:
+      replicas: 1
+      strategy: {}
+      template:
+        metadata:
+          creationTimestamp: null
+        spec:
+          containers:
+          - name: cds-events
+            resources:
+              limits:
+                cpu: 100m
+                memory: 128Mi
+              requests:
+                cpu: 100m
+                memory: 100Mi
+    status: {}
+  deps:  |
+    {
+      "dependencies": {
+        "request": "^2.81.0"
+      }
+    }
+
+  function:  |
+    var request = require('request');
+    module.exports = { main: function (event, context) {
+            console.log("data is" + JSON.stringify(event.data))
+            request({
+                url: "http://event-publish-service.kyma-system.svc.cluster.local:8080/v1/events",
+                method: "POST",
+                json: true,
+                body: event.data
+            }, function (error, response, body){
+                console.log(response);
+            });
+        } }
+  function-content-type: text
+  handler: handler.main
+  runtime: nodejs8
+  service:
+    ports:
+    - name: http-function-port
+      port: 8080
+      protocol: TCP
+      targetPort: 8080
+    selector:
+      created-by: kubeless
+      function: cds-events
+  timeout: ""
+---
+apiVersion: gateway.kyma-project.io/v1alpha2
+kind: Api
+metadata:
+  labels:
+    function: cds-events
+    example: cds-events
+  name: cds-events
+spec:
+  hostname: "cds-event-{{ .Release.Namespace }}.{{ .Values.cluster_domain}}"
```

Comment:
You expose the lambda without any authentication? And the lambda is putting the payload blindly in the event bus?
I do not see the relation to the oauth-proxy, when does that come into the game?

Reply:
1. For ease of configurability and being in preview mode, the lambda is not protected at the moment, there is another lambda on the CDS side which is preparing the appropriate payload. In the later iterations, I can add it.
2. To get the token from CDS, the OAuth credentials needs to be passed in the request body but the gateway passes them as header and also doesn't allow access to the body field. 
For that, we have created a proxy that takes the params and calls the token endpoint. The gateway then calls this proxy for token.